RFS/src/test/java/com/rfs/integration/WorkerCompleteAfterErrorTest.java

```diff
@@ -0,0 +1,118 @@
+    public void SingleWorkerFailure(final InjectedFailure failure){
+        // Setup
+        setupCms();
+        setupMockSource();
+        setupMockTarget();
+
+        // PSUEDO: Create a 5 workers
+        // PSUEDO: Wrap worker[0] with failure injector
+        // PSUEDO: Register worker[0] to fault on 'failure' parameter
+
+        // Action
+        // PSUEDO: Start all workers
+        // PSUEDO: Wait until all stop
+
+        // Verify
+        // PSUEDO: Verify 4 original workers finished without error
+        // PSUEDO: Verify 1 failure was injected
+        verifyTargetMigrated();
+    }
+
+    @ParameterizedTest(name = "Injected failure on {0} step")
+    @EnumSource(InjectedFailure.class)
+    public void ConcurrentAllWorkerFailure(){
+        // Setup
+        setupCms();
+        setupMockSource();
+        setupMockTarget();
+
+        // PSUEDO: Create a 5 workers
+        // PSUEDO: Wrap all workers with failure injector
+        // PSUEDO: Register all workers to fault on 'failure' parameter
+
+        // Action
+        // PSUEDO: Start all workers
+        // PSUEDO: Wait until all stop
+        // PSUEDO: Start 5 fresh workers
+
+        // Verification
+        // PSUEDO: Verify failure injector was called on all workers
+        // PSUEDO: Verify all new workers completed without issue
+        verifyTargetMigrated();
+    }
+
+    @Test
+    public void DisjointedAllWorkerFailure(){
+        // Setup
+        setupCms();
+        setupMockSource();
+        setupMockTarget();
+
+        final var workerCount = InjectedFailure.values().length + 1;
+        // PSUEDO: Create a workerCount workers
+        // PSUEDO: Wrap all workers with failure injector
+        // PSUEDO: Map InjectedFailure.values() onto workers of the same index
+
+        // Action
+        // PSUEDO: Start all workers
+        // PSUEDO: Wait until all stop
+
+        // Verification
+        // PSUEDO: Verify failure injector was called one once on each type of InjectedFailure
+        // PSUEDO: Verify last worker completed without issue
+        verifyTargetMigrated();
+    }
+
+    private void setupCms() {
+        // PSUEDO: Setup mock cms store
+    }
+
+    private void setupMockSource() {
+        // PSUEDO: Setup mock source cluster
+        //   - Mock snapshot workflow
+        //   - Mock index templates
+        //   - Mock indices
+        //   - Mock documents
+    }
+
+    private void setupMockTarget() {
+        // PSUEDO: Setup mock migration target
+        //   - Mock OpenSearch client
+    }
+
+    private void verifyTargetMigrated() {
+        // PSUEDO: Verify migration target
+        //   - Verify all index templates attempted at least once
+        //   - Verify all indices attempted at least once
+        //   - Verify all documents attempted at least once
+        // E.g. verify(targetClient, atLeastOnce()).createIndexTemplate("{index-template-name}", any())
+    }
+
+    private static enum InjectedFailure {
```

Comment:
I'd love to have tests that mimic RFS Workers acquiring a work lease and failing to complete the expected work before the lease times out.  Do you see that being a part of these tests, or a different set?

Reply:
I believe that this scenario would covered here - I imagine any fault will 'take down' the worker the other workers would 'pickup' the work.